### PR TITLE
Feature/auth with google id token

### PR DESCRIPTION
**Pull request by tomivm**
Allow Login with Firebase on Android and IOS

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -206,6 +208,46 @@ async function appleLogin(req, accessToken, refreshToken, idToken, profile, done
...
+      displayName: profile.name,
+      name: profile?.given_name,
+      lastname: profile?.family_name,
+      email: profile.email,
+      emails: [{ value: profile.email }],
+      photos: [{ value: profile.picture }]
+    };
+    await googleLogin(req, id_token, null, googleProfile, (error, response) => {
+      if (error) {
+        console.error(err);
```
Comment (RodriSanchez1) [inline api/controllers/user.js:238]: The parameter on `console.error` its wrong. Its `err` instead `error`

File: api/controllers/user.js [hunk 2]
```diff
@@ -206,6 +208,46 @@ async function appleLogin(req, accessToken, refreshToken, idToken, profile, done
...
+        res.redirect('/');
```
Comment (RodriSanchez1) [inline api/controllers/user.js:239]: Why do you use a redirect here? I think we can directly use  `res.status(404).json({message: "Something went wrong on Google Id Token login"});`

File: api/controllers/user.js [hunk 3]
```diff
@@ -206,6 +208,46 @@ async function appleLogin(req, accessToken, refreshToken, idToken, profile, done
...
+        return;
+      }
+      if (response.authToken) res.json(response);
+    });
+  } catch (err) {
+    console.error(err);
+    res.redirect('/');
```
Comment (RodriSanchez1) [inline api/controllers/user.js:246]: Same than above

**Review by RodriSanchez1 — CHANGES_REQUESTED** [review]

File: api/controllers/user.js [hunk 4]
```diff
@@ -206,6 +208,46 @@ async function appleLogin(req, accessToken, refreshToken, idToken, profile, done
   return passportLogin(ip, 'apple', accessToken, refreshToken, appleProfile, done);
 }
 
+async function googleIdTokenLogin(req, res) {
+  const client = new OAuth2Client();
+  const id_token = req.query.id_token;
+  async function verify() {
+    const ticket = await client.verifyIdToken({
+      idToken: id_token,
+      audience: [process.env.GOOGLE_FIREBASE_SIGN_IN_APP_ID, process.env.GOOGLE_FIREBASE_WEB_CLIENT_SIGN_IN]
```
Comment (RodriSanchez1) [inline api/controllers/user.js:217]: change `process.env.GOOGLE_FIREBASE_WEB_CLIENT_SIGN_IN` to `process.env.GOOGLE_FIREBASE_WEB_CLIENT_ID` please :)

**Review by martinbedouret — APPROVED** [review]

**Pull request merged**